sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -142,6 +197,13 @@ def test_check_estimator():
     msg = "it does not implement a 'get_params' methods"
     assert_raises_regex(TypeError, msg, check_estimator, object)
     assert_raises_regex(TypeError, msg, check_estimator, object())
+    # check that values returned by get_params match set_params
+    msg = "get_params result does not match what was passed to set_params"
+    assert_raises_regex(AssertionError, msg, check_estimator,
+                        ModifiesValueInsteadOfRaisingError())
+    check_estimator(RaisesError())
+    assert_raises_regex(AssertionError, msg, check_estimator,
```

Comment:
can you maybe run the test on a pipeline to make sure it works on deep estimators?